src/target/riscv/riscv-013.c

```diff
@@ -3058,6 +3058,21 @@ void riscv013_clear_abstract_error(struct target *target)
 	total_tests++;		    \
 }
 
+#define COMPLIANCE_MUST_PASS(b) COMPLIANCE_TEST(ERROR_OK == (b), "Regular calls must return ERROR_OK")
+
+#define COMPLIANCE_READ(target, addr, value) COMPLIANCE_MUST_PASS(dmi_read(target, addr, value))
+#define COMPLIANCE_WRITE(target, addr, value) COMPLIANCE_MUST_PASS(dmi_write(target, addr, value))
+
+#define COMPLIANCE_CHECK_RO(target, addr)                               \
```

Comment:
Can this be a function?

Reply:
well, not as currently written because the passed_test macro is a local variable (yes its not beautiful). So then I can't use any of the macros.

Reply:
Alright. I'm not a fan of macros, but if you like it this way that's good enough.